### DISCUSSION TITLE
Undefined array key "code" ketika $bundle->addCondition($conditionFh);

### DISCUSSION THREAD

**Discussion by afifsyaifullah1512** [ivanwilliammd/satusehat-integration#124] [Q&A (Tanya Jawab)] [answered]
Ketika proses post bundle, saya mendapatkan error Undefined array key "code" yang kalau dilihat errornya berasal dari library Encounter->addCondition

![WhatsApp Image 2024-04-01 at 21 33 45_3b2b06e2](https://github.com/ivanwilliammd/satusehat-integration/assets/51071756/6cd86f33-54b0-4215-a656-aea70f510d5f)

terus ketika saya debug dari library Encounter->addCondition dengan perintah var_dump($condition) muncul result seperti ini dan ada notice bahwa “Unauthorized. Token not found”

![WhatsApp Image 2024-04-01 at 21 34 21_01a7e045](https://github.com/ivanwilliammd/satusehat-integration/assets/51071756/db8ce897-a297-4734-a86f-0573efde81b6)

Undefined array key "code" berasar dari code ini
![WhatsApp Image 2024-04-01 at 21 34 52_dc325cf4](https://github.com/ivanwilliammd/satusehat-integration/assets/51071756/afa7cd9f-eb9f-4476-8b1a-0f2e05e3f4be)

ini penggunaan library bundle pada controller saya

![WhatsApp Image 2024-04-01 at 22 23 44_d7ab9d1d](https://github.com/ivanwilliammd/satusehat-integration/assets/51071756/4f222f60-b94d-4e1e-82da-31b4c8658f6c)

kira-kira salahnya ada di codingan penerapan bundle atau gimana ya dok?

**Comment by ivanwilliammd** [comment]
Oauth2_error itu hanya error handling yg ada di dalam class $condition

Terkait dengan error code, sy curiga ini karena ada indexing yg ditambahkan --> pls kindly check / confirm @SyaefulKai bagian ini

```$condition->condition[code][coding][0][code]```

Btw ini sdh composer update dengan library versi terbaru kah 2.3.1 @afifsyaifullah1512 ?

Kindly note : itu client_id & client_secret nya muncul... Lain kali disensor dahulu @afifsyaifullah1512

**Comment by afifsyaifullah1512** [comment]
Sudah composer yang baru dok.

Ohiya terima kasih udah mengingatkan dok, lupa sensor, sudah saya update gambarnya untuk disensor.

**Comment by SyaefulKai** [comment]
Saya barusan cek.

Error undefined array key "code" kemungkinan bisa terjadi karena kode ICD10 yang anda masukan itu tidak ditemukan dalam database. Ini hasil ketika kode ICD10 tidak ditemukan dalam database.
<img width="687" alt="bundleerror1" src="https://github.com/ivanwilliammd/satusehat-integration/assets/158563298/47790365-21b8-43d2-8726-e3703bbe59b8">

Coba pastikan kode ICD10 yang Anda masukan itu valid. Ini hasil ketika kode ICD10 ditemukan.
<img width="574" alt="bundleerror2" src="https://github.com/ivanwilliammd/satusehat-integration/assets/158563298/e4c0b259-b059-400f-a926-f5ad44534520">

**Comment by ivanwilliammd** [comment]
Mantap @SyaefulKai .. 

Apakah bisa dibantu untuk PR jadi improvement di Condition addCode @SyaefulKai ?

Passing code dahulu dengan model

Icd10::where('code', $.....)->get() --> bila count > 0 baru proses, bila tidak return error / throw exception?

**Comment by SyaefulKai** [comment]
Bisa pak

**Comment by afifsyaifullah1512** [comment]
Terimakasih banyak Dok @ivanwilliammd dan mas @SyaefulKai untuk informasi dan updatenya, saya coba dulu.

**Comment by afifsyaifullah1512** [comment]
Selamat malam dok, untuk error Undefined array key "code" sudah fix karena saya lupa melakukan seed ke tabel icd10. Tetapi muncul error baru yang mengarah ke $bundle->post seperti gambar ini
![7a7cff66-e0f8-4c8d-b5ec-0b369052f4b7](https://github.com/ivanwilliammd/satusehat-integration/assets/51071756/88fbe697-2f6e-48aa-834b-c4c3ff5cb20b)


saya coba check apakah data array $encounter & $condition kosong, ternyata ketika di debug datanya ada, seperti gambar dibawah
![d97ee593-93c5-43e2-bb6d-f9f75634776f](https://github.com/ivanwilliammd/satusehat-integration/assets/51071756/7b2a6b89-4302-4140-9242-ccdb5d4fcf7a)


kira-kira salahnya dimana ya dok?

@ivanwilliammd

**Comment by ivanwilliammd** [comment]
debug-nya sprtinya bukan ke $bundleFh
Mgkn saat dev kmrn apakah ada temuan ini @SyaefulKai ?

**Comment by SyaefulKai** [comment]
Invalid Resource Type: array terjadi karena FHIR bundle mengirim $this->bundle tanpa melalui proses json_encode. Jadinya masih bertipe array, sedangkan yang programnya harapkan adalah tipe JSON.
Sebelum melalui proses json_encode:
<img width="630" alt="bundle4" src="https://github.com/ivanwilliammd/satusehat-integration/assets/158563298/28342db5-8b2e-418b-9d73-9082b9167d49">

Sesudah:
<img width="701" alt="bundlerror3" src="https://github.com/ivanwilliammd/satusehat-integration/assets/158563298/05a66de7-e4c6-4c6b-a811-0023cf979fe1">

**Comment by afifsyaifullah1512** [comment]
berarti sebelum di post ke bunde harus di json_encode terlebih dahulu mas? @SyaefulKai

**Comment by ivanwilliammd** [comment, marked as answer]
Barusan sudah saya hotfix di v2.3.3 @afifsyaifullah1512 

![image](https://github.com/ivanwilliammd/satusehat-integration/assets/48039956/1ba0515b-9abd-4a9f-b41d-f1495eea509e)


sekarang cukup $bundleFh->post() saja
